src/search/pdbs/pattern_database.cc

```diff
@@ -1,360 +1,20 @@
-          the same abstract transition and randomly pick one of them to
-          set for the next state. We iterate until reaching a goal state.
-          Note that this kind of plan extraction does not uniformly at random
-          consider all successor of a state but rather uses the arbitrarily
-          chosen generating operator to settle on one successor state, which
-          is biased by the number of operators leading to the same successor
-          from the given state.
-        */
-        State initial_state = task_proxy.get_initial_state();
-        initial_state.unpack();
-        int current_state =
-            hash_index(initial_state.get_unpacked_values());
-        if (distances[current_state] != numeric_limits<int>::max()) {
-            while (!is_goal_state(current_state, abstract_goals, variables)) {
-                int op_id = generating_op_ids[current_state];
-                assert(op_id != -1);
-                const AbstractOperator &op = operators[op_id];
-                int successor_state = current_state - op.get_hash_effect();
-
-                // Compute equivalent ops
-                vector<OperatorID> cheapest_operators;
-                vector<int> applicable_operator_ids;
-                match_tree.get_applicable_operator_ids(successor_state, applicable_operator_ids);
-                for (int applicable_op_id : applicable_operator_ids) {
-                    const AbstractOperator &applicable_op = operators[applicable_op_id];
-                    int predecessor = successor_state + applicable_op.get_hash_effect();
-                    if (predecessor == current_state && op.get_cost() == applicable_op.get_cost()) {
-                        cheapest_operators.emplace_back(applicable_op.get_concrete_op_id());
-                    }
-                }
-                if (compute_wildcard_plan) {
-                    rng->shuffle(cheapest_operators);
-                    wildcard_plan.push_back(move(cheapest_operators));
-                } else {
-                    OperatorID random_op_id = *rng->choose(cheapest_operators);
-                    wildcard_plan.emplace_back();
-                    wildcard_plan.back().push_back(random_op_id);
-                }
-
-                current_state = successor_state;
-            }
-        }
-        utils::release_vector_memory(generating_op_ids);
-    }
-}
-
-bool PatternDatabase::is_goal_state(
-    int state_index,
-    const vector<FactPair> &abstract_goals,
-    const VariablesProxy &variables) const {
-    for (const FactPair &abstract_goal : abstract_goals) {
-        int pattern_var_id = abstract_goal.var;
-        int var_id = pattern[pattern_var_id];
-        VariableProxy var = variables[var_id];
-        int temp = state_index / hash_multipliers[pattern_var_id];
-        int val = temp % var.get_domain_size();
-        if (val != abstract_goal.value) {
-            return false;
-        }
-    }
-    return true;
+    Pattern &&pattern,
```

Comment:
I liked the old interface better because it prevented some wrong usage of the class. With the new interface, the values of `pattern`, `num_states`, `distances`, and `hash_multipliers` could disagree. So as a user of the class, once you pick the first constructor parameter, you have to put specific values in the other parameters. I see forced choices in parameters as a bit of a code smell but I don't have a better suggestion that also moves the computation out of the constructor..